addons/mail/models/template_reset_mixin.py

```diff
@@ -0,0 +1,142 @@
+                    if base_lang_code not in processed_base_langs:
+                        processed_base_langs.append(base_lang_code)
+                        self._process_translation_data(base_trans_file, code, xml_ids)
+
+            # Step 2: reset translation file with main language file (can possibly override the
+            # terms coming from the base language)
+            trans_file = get_module_resource(module_name, 'i18n', lang_code + '.po')
+            if trans_file:
+                is_po_file_available = True
+                self._process_translation_data(trans_file, code, xml_ids)
+
+            # If no translation file available, unlink custom translated terms linked to template
+            if not is_po_file_available:
+                trans_to_remove.append(code)
+
+        if trans_to_remove:
+            self.env['ir.translation'].search([
+                ('name', 'like', f'{self._name},%'),
+                ('res_id', '=', self.id),
+                ('lang', 'in', trans_to_remove),
+                ('module', 'in', [module_name, False]),
+            ]).unlink()
+
+    def _process_translation_data(self, trans_file, lang, xml_ids):
+        """Populates the ir_translation table.
+        :param trans_file: path to a translation file to open, e.g. {addon_path}/mail/i18n/es.po
+        :param lang: language code of the translations contained in `trans_file`
+                     language must be present and activated in the database
+        """
+        with file_open(trans_file, mode='rb', filter_ext=(".po",)) as fileobj:
+            fileformat = os.path.splitext(trans_file)[-1][1:].lower()
+            reader = TranslationFileReader(fileobj, fileformat=fileformat)
+            # Process a single PO entry
+            for row in reader:
+                if row.get('imd_name') in xml_ids:
+                    # copy Translation from Source to Destination object
+                    self.env['ir.translation']._set_ids(
+                        row['name'], 'model', lang, self._ids, row.get('value', ''), row['src'],
+                    )
+
+    def reset_template(self):
+        """Resets the Template with values given in source file. We ignore the case of
+        template being overridden in another modules because it is extremely less likely
+        to happen. This method also tries to reset the translation terms for the current
+        user lang (all langs are not supported due to costly file operation). """
+        for template in self.filtered(lambda t: t.template_fs and t.is_template_modified):
+            external_id = template.get_external_id().get(template.id)
+            module, xml_id = external_id.split('.')
+            fullpath = get_resource_path(*template.template_fs.split('/'))
+            if fullpath:
+                doc = etree.parse(fullpath)
+                for rec in doc.xpath("//record[@id='%s'] | //record[@id='%s']" % (external_id, xml_id)):
```

Comment:
FWIW lxml suports xpath variables, which I assume (though have not tested) it escapes properly, you can pass the variables as kwargs and refer to them using `$<name>` inside the query:
```python
>>> expr = "//*[local-name() = $name]"
>>> print(root.xpath(expr, name = "foo")[0].tag)
foo
>>> print(root.xpath(expr, name = "bar")[0].tag)
bar
```